yt_dlp/extractor/boosty.py

```diff
@@ -160,18 +163,39 @@ def _extract_formats(self, player_urls, video_id):
                 self.report_warning(f'Unknown format type: {format_type!r}')
         return formats
 
+    def _extract_posts_from_html(self, url, post_id, webpage=None):
+        if webpage is None:
+            webpage = self._download_webpage(url, video_id=post_id)
+        initial_state_str = self._search_regex(
+            r'(?s)<script[^>]+id=[\'"]initial-state[\'"][^>]*>([^<]+)</script>',
+            webpage, 'initial-state', fatal=False)
+        try:
+            initial_state = json.loads(initial_state_str)
+            posts_list = initial_state['posts']['postsList']['data']['posts']
+        except (JSONDecodeError, KeyError):
+            self.report_warning('failed to extract posts from website')
+            return []
+        return posts_list
+
     def _real_extract(self, url):
         user, post_id = self._match_valid_url(url).group('user', 'post_id')
         post = self._download_json(
             f'https://api.boosty.to/v1/blog/{user}/post/{post_id}', post_id,
             note='Downloading post data', errnote='Unable to download post data')
+        all_posts = [post]
 
         post_title = post.get('title')
+        webpage = None
         if not post_title:
             self.report_warning('Unable to extract post title. Falling back to parsing html page')
             webpage = self._download_webpage(url, video_id=post_id)
             post_title = self._og_search_title(webpage, default=None) or self._html_extract_title(webpage)
 
+        private = bool(post.get('subscriptionLevel'))
```

Comment:
FWIW, there is a boolean field `hasAccess`, which is always `true` for public posts and `false` for paywalled posts if a) you are no authenticated, or b) you don't have a required subscription level.